source/plugins/ruby/extension_utils.rb

```diff
@@ -2,26 +2,112 @@
+      end
+      return outputStreamId
     end
+
+    def isAADMSIAuthMode()
+      return !ENV["AAD_MSI_AUTH_MODE"].nil? && !ENV["AAD_MSI_AUTH_MODE"].empty? && ENV["AAD_MSI_AUTH_MODE"].downcase == "true"
+    end
+
+    def isDataCollectionSettingsConfigured
+      isCollectionSettingsEnabled = false
+      begin
+        dataCollectionSettings = Extension.instance.get_extension_data_collection_settings()
+        if !dataCollectionSettings.nil? && !dataCollectionSettings.empty?
+          isCollectionSettingsEnabled = true
+        end
+      rescue => errorStr
+        $log.warn("ExtensionUtils::isDataCollectionSettingsConfigured: failed with an exception: #{errorStr}")
+      end
+      return isCollectionSettingsEnabled
+    end
+
+    def getDataCollectionIntervalSeconds
+      collectionIntervalSeconds = 60
+      begin
+        dataCollectionSettings = Extension.instance.get_extension_data_collection_settings()
+        if !dataCollectionSettings.nil? &&
+           !dataCollectionSettings.empty? &&
+           dataCollectionSettings.has_key?(Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_INTERVAL)
+          interval = dataCollectionSettings[Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_INTERVAL]
+          re = /^[0-9]+[m]$/
+          if !re.match(interval).nil?
+            intervalMinutes = interval.dup.chomp!("m").to_i
+            if intervalMinutes.between?(Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_INTERVAL_MIN, Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_INTERVAL_MAX)
+              collectionIntervalSeconds = intervalMinutes * 60
+            else
+              $log.warn("ExtensionUtils::getDataCollectionIntervalSeconds: interval value not in the range 1m to 30m hence using default, 60s: #{errorStr}")
+            end
+          else
+            $log.warn("ExtensionUtils::getDataCollectionIntervalSeconds: interval value is invalid hence using default, 60s: #{errorStr}")
+          end
+        end
+      rescue => errorStr
+        $log.warn("ExtensionUtils::getDataCollectionIntervalSeconds: failed with an exception: #{errorStr}")
+      end
+      return collectionIntervalSeconds
+    end
+
+    def getNamespacesForDataCollection
+      nameSpaces = []
+      begin
+        dataCollectionSettings = Extension.instance.get_extension_data_collection_settings()
+        if !dataCollectionSettings.nil? &&
+           !dataCollectionSettings.empty? &&
+           dataCollectionSettings.has_key?(Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACES)
+          nameSpacesSetting = dataCollectionSettings[Constants::EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACES]
+          if !nameSpacesSetting.nil? && !nameSpacesSetting.empty? && nameSpacesSetting.kind_of?(Array) && nameSpacesSetting.length > 0
+            uniqNamespaces = nameSpacesSetting.uniq
+            nameSpaces = uniqNamespaces.map(&:downcase)

```

Comment:
maybe add a check for namespaces? 

From: https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace 
The name of your namespace must be a valid DNS label. (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names)

Reply:
> maybe add a check for namespaces?
> 
> From: https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace The name of your namespace must be a valid DNS label. (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names)

> namespace constraints being validated at the client & amcs side so that customer can get the feedback immediately at the ARM layer which would avoid customer to look at the logs inside the container and figure out.